Snowflake.Data/Logger/EasyLoggingLevelMapper.cs

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+using System;
+using log4net.Core;
+using Snowflake.Data.Configuration;
+
+namespace Snowflake.Data.Log
+{
+    internal class EasyLoggingLevelMapper
+    {
+        private static EasyLoggingLevelMapper s_instance = null;
+
+        private static readonly object s_lock = new object();
+
+        public static EasyLoggingLevelMapper Instance
+        {
+            get
+            {
+                lock (s_lock)
+                {
+                    if (s_instance == null)
+                    {
+                        s_instance = new EasyLoggingLevelMapper();
+                    }
+
+                    return s_instance;
+                }
+            }
+        }
+        
+        public Level ToLog4NetLevel(EasyLoggingLogLevel level)
```

Comment:
can we add this method on the `EasyLoggingLogLevel` enum? then we won't need this mapper at all

Reply:
I would rather not. EasyLogingLevel is like a generic class and log4net.Core.Level is like a more specific implementation. It is better if api doesn't know about the implementation. If in future we would substitute log4net library by something else then EasyLoggingLevel enum will not have a reason to change.